test/e2e/specs/editor/blocks/list.spec.js

```diff
@@ -0,0 +1,872 @@
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Create a list with the slash block shortcut.
+		await page.locator( 'role=button[name="Add default block"i]' ).click();
+		await page.keyboard.type( 'test' );
+		await pageUtils.pressKeyTimes( 'ArrowLeft', 4 );
+		await page.keyboard.type( '* ' );
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:list -->
+<ul><li>test</li></ul>
+<!-- /wp:list -->`
+		);
+	} );
+
+	test( 'can be created by using a number at the start of a paragraph block', async ( {
+		editor,
+		page,
+	} ) => {
+		// Create a block with some text that will trigger a list creation.
+		await page.locator( 'role=button[name="Add default block"i]' ).click();
+		await page.keyboard.type( '1) A list item' );
+
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:list {"ordered":true} -->
+<ol><li>A list item</li></ol>
+<!-- /wp:list -->`
+		);
+	} );
+
+	test( 'can undo asterisk transform', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await page.locator( 'role=button[name="Add default block"i]' ).click();
+		await page.keyboard.type( '1. ' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'z' );
+
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:paragraph -->
+<p>1. </p>
+<!-- /wp:paragraph -->`
+		);
+	} );
+
+	test( 'should undo asterisk transform with backspace', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.locator( 'role=button[name="Add default block"i]' ).click();
+		await page.keyboard.type( '* ' );
+		await page.keyboard.press( 'Backspace' );
+
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:paragraph -->
+<p>* </p>
+<!-- /wp:paragraph -->`
+		);
+	} );
+
+	test( 'should undo asterisk transform with backspace after selection changes', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.locator( 'role=button[name="Add default block"i]' ).click();
+		await page.keyboard.type( '* ' );
+		await page.evaluate( () => new Promise( window.requestIdleCallback ) );
+		await page.keyboard.press( 'Backspace' );
+
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:paragraph -->
+<p>* </p>
+<!-- /wp:paragraph -->`
+		);
+	} );
+
+	test( 'should undo asterisk transform with backspace setting isTyping state', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.locator( 'role=button[name="Add default block"i]' ).click();
+		await page.keyboard.type( '* ' );
+		await editor.showBlockToolbar();
+		await page.keyboard.press( 'Backspace' );
+
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:paragraph -->
+<p>* </p>
+<!-- /wp:paragraph -->`
+		);
+	} );
+
+	test( 'should undo asterisk transform with backspace after selection changes without requestIdleCallback', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.locator( 'role=button[name="Add default block"i]' ).click();
+		await page.evaluate( () => delete window.requestIdleCallback );
```

Comment:
Why?

Reply:
@kevin940726 

https://github.com/WordPress/gutenberg/pull/41555/files#diff-128f270bb6aa4c6992e726c44593f7cd70c7b3638c8933e76914aaa2eccaa6f2L85-L93

I think this test is for environments that do not have requestIdleCallback.

Reply:
I see. Now that we're using Playwright, I think we can just remove this and enable running Safari for the test suite.

However, we don't have the necessary architecture for that yet. So I guess we can let this pass for now.

This code should probably stay in unit tests though, not sure why it's in e2e tests.